Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -73,140 +73,184 @@ class WalletConnectManager {
                             }
                         }
                     }
-
+                    
                     if let dappConnectedTrackingEvent = dappConnectedTrackingEvent, !session.peer.name.isEmpty {
                         let dappName = session.peer.name.prefix(100)
                         Tracker.trackEvent(dappConnectedTrackingEvent, parameters: ["dapp_name": dappName])
                     }
                 }
-
+                
                 dappConnectedTrackingEvent = nil
                 NotificationCenter.default.post(name: .wcDidConnectSafeServer, object: self)
             }.store(in: &publishers)
-
-        Sign.instance.sessionDeletePublisher
+        
+        Web3Wallet.instance.sessionDeletePublisher
             .receive(on: DispatchQueue.main)
-            .sink { [unowned self] response in
-                deleteStoredSession(topic: response.0)
+            .sink { [unowned self] (topic, reason) in
+                deleteStoredSession(topic: topic)
                 NotificationCenter.default.post(name: .wcDidDisconnectSafeServer, object: self)
             }.store(in: &publishers)
     }
-
+    
     func canConnect(url: String) -> Bool {
         WalletConnectURI(string: url) != nil
     }
-
+    
     func pairClient(url: String, trackingEvent: TrackingEvent?) {
         guard let uri = WalletConnectURI(string: url) else { return }
         dappConnectedTrackingEvent = trackingEvent
         pairClient(uri: uri)
     }
-
+    
     func pairClient(uri: WalletConnectURI) {
         Task {
             do {
-                try await Sign.instance.pair(uri: uri)
+                // Clean up pairings to prevent pairingAlreadyExist error
+                disconnectUnusedPairings()
```

Comment:
This is the place where it shouldn't be necessary to clean out old pairings. I am also not sure whether it prevents pairingAlreadyExist errors.